### PR TITLE
Monitor feature implemented using decorator design pattern

### PR DESCRIPTION
**Pull request by FawBiotechnologist**
This is our first proposition, let us know if that's what you were asking for.

### CODE REVIEW

File: src/main/java/edu/kis/powp/jobs2d/drivers/decorators/Job2dDriverMonitorDecorator.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+		setStartingPos(x, y);
+	}
+
+	private void setStartingPos(int x, int y) {
+		startX = x;
+		startY = y;
+	}
+
+	// ze wzoru na odleglosc miedzy dwoma punktami
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/decorators/Job2dDriverMonitorDecorator.java]: komentarz zbędny

File: src/main/java/edu/kis/powp/jobs2d/drivers/decorators/Job2dDriverMonitorDecorator.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+import edu.kis.powp.jobs2d.Job2dDriver;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public final class Job2dDriverMonitorDecorator extends BaseJob2dDriverDecorator {
+	private final static Logger logger = Logger.getLogger(Logger.GLOBAL_LOGGER_NAME);
+	private double totalDistanceTraveled;
+	private double operationalDistanceTraveled;
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/decorators/Job2dDriverMonitorDecorator.java]: Add getters to make the usage monitor potentially usable.

File: src/main/java/edu/kis/powp/jobs2d/drivers/decorators/Job2dDriverMonitorDecorator.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
+package edu.kis.powp.jobs2d.drivers.decorators;
+
+import edu.kis.powp.jobs2d.Job2dDriver;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public final class Job2dDriverMonitorDecorator extends BaseJob2dDriverDecorator {
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/decorators/Job2dDriverMonitorDecorator.java]: I suggest Job2dDriverUsageMonitorDecorator

**Review by kowallus — CHANGES_REQUESTED** [review]
Some minor issues.

**Comment by FawBiotechnologist** [comment]
I fixed all mentioned issues in separate commits. Re-review, and let us know if there are still any issues to fix.

**Pull request merged**